src/services/solana-client/transferWAudioBalance.ts

```diff
@@ -0,0 +1,172 @@
+
+  // hash the recipient solana pubkey and create signature
+  const msgHash = keccak256(recipeintPubkey.toBytes())
+  const signatureObj = secp256k1.ecdsaSign(
+    Uint8Array.from(msgHash),
+    ethPrivateKeyArr
+  )
+
+  const instructionData = new TransferInstructionData({
+    signature: signatureObj.signature,
+    recoveryId: signatureObj.recid,
+    ethAddress: ethAddressArr
+  })
+
+  // serialize it
+  const serializedInstructionData = borsh.serialize(
+    transferInstructionSchema,
+    instructionData
+  )
+
+  // give it the rust enum tag
+  // we can do this better all with borsh - look in
+  // identity or ask cheran
+  const serializedInstructionEnum = Uint8Array.of(
+    1,
+    ...serializedInstructionData
+  )
+
+  const connection = new Connection(solanaClusterEndpoint)
+
+  const accounts: AccountMeta[] = [
+    // sender account (bank)
+    { pubkey: senderSolanaPubkey, isSigner: false, isWritable: true },
+    // receiver token account
+    { pubkey: recipeintPubkey, isSigner: false, isWritable: true },
+    // Bank token account authority (PDA of the userbank program)
+    { pubkey: generatedProgramPDA, isSigner: false, isWritable: false },
+    // spl token account
+    {
+      pubkey: tokenProgramKey,
+      isSigner: false,
+      isWritable: false
+    },
+    // sysvar instruction id
+    {
+      pubkey: SYSVAR_INSTRUCTIONS_PUBKEY,
+      isSigner: false,
+      isWritable: false
+    }
+  ]
+
+  // const instr = new TransactionInstruction({
+  //   keys: accounts,
+  //   programId: audiusProgramPubkey,
+  //   data: Buffer.from(serializedInstructionEnum)
+  // })
+
+  // eth pubkey is different from the ethAddress - addresses are len 20, pub keys are len 64
+  const ethPubkey = secp256k1.publicKeyCreate(ethPrivateKeyArr, false).slice(1)
+  // this particular program expects there to be an secp recovery instruction
+  // prior to the claim instruction, so make it and add it
+  // const secpInstruction = Secp256k1Program.createInstructionWithPublicKey({
+  //   publicKey: ethPubkey,
+  //   message: recipeintPubkey.toBytes(),
+  //   signature: signatureObj.signature,
+  //   recoveryId: signatureObj.recid
+  // })
+  console.log({ ethPubkey })
+  const { blockhash } = await connection.getRecentBlockhash()
+
+  const transactionData = {
+    recentBlockhash: blockhash,
+    secpInstruction: {
+      publicKey: Buffer.from(ethPubkey),
+      message: recipeintPubkey.toString(),
+      signature: Buffer.from(signatureObj.signature),
+      recoveryId: signatureObj.recid
+    },
+    instruction: {
+      keys: accounts.map(account => {
+        return {
+          pubkey: account.pubkey.toString(),
+          isSigner: account.isSigner,
+          isWritable: account.isWritable
+        }
+      }),
+      programId: audiusProgramPubkey.toString(),
+      data: Buffer.from(serializedInstructionEnum)
+    }
+  }
+
+  const response = await SolanaClient.relay(transactionData)
```

Comment:
Not sure I love this pattern of importing the SolanaClient on a method we attach to the Solana Client - seems like this makes our life a bit harder if we wanted to test this function while mocking the relay for instance?